### PR TITLE
#443 Feature - gui scrollback

### PR DESCRIPTION
**Pull request by Doverstav**
#443 

Enormous pull request containing the MVP for the scrollback feature. Check that everything works on your machine and feels reasonably good to use, and we'll refine the experience in the coming sprints.

### CODE REVIEW

File: src/js/view/components/LogViewer.js [hunk 1]
```diff
@@ -103,14 +158,98 @@ const LogViewer = props => {
     });
   }, [props.source.path]);
 
+  useEffect(() => {
+    const wheelHandler = event => {
```
Comment (gustavjorlov) [inline src/js/view/components/LogViewer.js]: What is a wheel 🤔

Comment (Doverstav) [reply src/js/view/components/LogViewer.js]: image

Pierre and his ideas... (It's actually a mousewheel)

But yeah, naming could probably be clearer if that's what you're getting at!

Comment (MelindaSW) [reply src/js/view/components/LogViewer.js]: How does 'handleScrollWheelEvent' sound to you? Is it still too vague?

File: src/js/view/actions/dispatchActions.js [hunk 2]
```diff
@@ -74,12 +74,20 @@ export const updateLastSeenLogSizes = (dispatch, path, fileSize) => {
   });
 };
 
-export const setFileData = (dispatch, filePath, fileSize, history) => {
+export const setFileData = (
+  dispatch,
+  filePath,
+  fileSize,
+  history,
+  metaData
```
Comment (MelindaSW) [inline src/js/view/actions/dispatchActions.js]: metaData feels a little vague. We are not really using it for now since we did not need it for the MVP at the moment. It will probably be useful though to keep for the coming issues. How about changing it to `logLinesMetaData { startingByteOfLines: { [sourcePath]: 12345  } }` or something to the like?

File: src/js/adapters/fileReader.js [hunk 3]
```diff
@@ -301,8 +310,11 @@ const readDataFromByte = (filePath, start, numberOfBytes) => {
 };
 
 const parseSelectedData = (data, start, numberOfBytes) => {
```
Comment (MelindaSW) [inline src/js/adapters/fileReader.js]: I would wish to have parseSelectedData renamed to something that connects it more to the readDataFromByte workflow. Perhaps parseDataReadFromBytePositionInFile or parseDataFromBytePositionIntoLineArrayAndCreateReturnObject ?

Comment (Doverstav) [reply src/js/adapters/fileReader.js]: Hmm, while I agree that the name isn't the best at the moment (and I don't have a better suggestion), the suggested names feel too wordy and perhaps overly specific? In my opinion, a name should represent what a function does independently, and not in relation to other things around.   

The last suggestion might be on to something though, perhaps something like `parseByteDataIntoStringArray` is a good name, which describes what it does fairly accurately. Alternatively, in order to make it clear that metadata is also calculated, you could instead call it `parseByteDataIntoStringArrayWithMetadata`. Documentation of the return object might be done with comments, such as those documenting `parseLinesBackwards`.

Although, now that I'm writing all this out, I am realizing that this function should probably be split up to make things even clearer. One function shoudl be responsible for converting to a string array ( `parseByteDataIntoStringArray` ), and one should extract metdata ( `extractMetadataFromByteData` ), and perhaps single function to combine these into an object to return (I don't have good name for this one yet).

Comment (MelindaSW) [reply src/js/adapters/fileReader.js]: Totally agree on the wordiness. The splitting into several functions sounds like a good idea.
